src/main/java/com/desiato/puresynth/services/SessionService.java

```diff
@@ -6,22 +6,36 @@
 import com.desiato.puresynth.repositories.SessionRepository;
 import com.desiato.puresynth.repositories.UserRepository;
 import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @AllArgsConstructor
+@Slf4j
 @Service
 public class SessionService {
 
     private final SessionRepository sessionRepository;
     private final UserRepository userRepository;
 
     public Session createSession(User user) {
+        if (user.getEmail() == null) {
```

Comment:
When can it happen? This check doesn't provide any value. A classic `NPE` also gives the same result.

Reply:
yes, being already in place early checks, this is not useful.